src/constraints.jl

```diff
@@ -0,0 +1,84 @@
+export
+    run_constraints!,
+    SHAKE,
+    apply_constraint!
+
+
+
+function get_masses(sys)
+    masses = []
+    for atom in sys.atoms
+        push!(masses, atom.mass)
+    end
+    return masses
+end
+
+function run_constraints!(sys, new_coords, dt)
+    if length(sys.constraints) == 0
+        sys.coords = new_coords
+    end
+
+    for cons in sys.constraints
+        apply_constraint!(sys, new_coords, dt, cons)
+    end
+end
+
+
+
+struct SHAKE{D, B} <: Constraint
+    d::D
+    bond_list::B
+end
+
+
+function apply_constraint!(sys, new_coords, dt, constr::SHAKE)
+    
+    bond_list = constr.bond_list
```

Comment:
These setup steps that stay the same throughout the simulation can be moved to the `SHAKE` constructor and stored in the struct.